### ISSUE TITLE
Homebrew doesn't install correctly: "Fatal: Could not resolve HEAD to a revision"

### ISSUE THREAD

**Issue by Renee-Napoliello** [Homebrew/homebrew-core#69265]
# Bug report

**Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.**

- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [x] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [x] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

## What you were trying to do (and why)

I am trying to install and use homebrew.

## What happened (include command output)

I installed homebrew. During installation, the message "Fatal: Could not resolve HEAD to a revision" appeared. However, it also said "installation successful!" Next, I attempted to install various packages, including cask, wget, hello, htop, and nmap. For every attempt the message "Fatal: Could not resolve HEAD to a revision" appeared and the package failed to install. I uninstalled homebrew and reinstalled it three times. The same error appears. 
<details>
  <summary>Command output</summary>
  <pre>

  <fatal: Could not resolve HEAD to a revision>

  </pre>
</details>

## What you expected to happen

< installation successful>

## Step-by-step reproduction instructions (by running `brew install` commands)

</bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)">

**Comment by Renee-Napoliello** [comment]
Right now I **am concerned I somehow corrupted homebrew permanently** on my compter, and it is unresolvable unless I get a new computer. In a previous installation attempted, _I screwed up royally._ I ran a command in the _middle_ of installing homebrew because I though homebrew was finished installing. Then, realizing my mistake,  I interrupted and terminated installation. I reopened the terminal and re-started installation. Perhaps I corrupted homebrew in this error.
